main.js

```diff
@@ -1,147 +1,64 @@
-      .send(htmlTemplate`
-        <h1>Erreur</h1>
-        <p>le paramètre <code>destination<code> est manquant.</p>
-        <p>Il est sûrement manquant en tant que paramètre du <code>redirect_uri</code> du lien "login with github"
-      `)
-    return;
-  }
-
-  const redirectUrl = new URL(destination)
-  const hostname = redirectUrl.hostname
-
-  if(!hostname){
-    res.status(400)
-      .header('Content-Type', 'text/html')
-      .send(htmlTemplate(`
-        <h1>Erreur</h1>
-        <p>le paramètre <code>destination<code> n'a pas de hostname. (destination : ${destination})</p>
-        <p>Rajouter une origine au paramètre <code>destination<code>
-      `))
-    return;
-  }
+if(process.env.GITHUB_OAUTH_APP_CLIENT_ID){
+  if(!process.env.GITHUB_OAUTH_APP_CLIENT_SECRET){
+    console.error(`
+      Une application GitHub est configurée sans secret.
 
-  if(hostname !== 'localhost' && !allowlist.has(hostname)){
-    res.status(403)
-      .header('Content-Type', 'text/html')
-      .send(htmlTemplate(`
-        <h1>Erreur</h1>
-        <p>La destination est ${destination}, et son hostname (${hostname}) n'est pas présent dans notre <a href="https://github.com/Scribouilli/toctoctoc/blob/main/allowlist.csv">liste de hostname autorisés</a>.</p>
-        <p>Liste des hostname autorisés : 
-          <ul>
-            ${[...allowlist].map(hostname => `<li>${hostname}</li>`).join('')}
-          </ul>
-        </p>
-        <p>Changer cette liste ou installez une nouvelle instance de toctoctoc où ce hostname est autorisé</p>
-      `))
-    return;
+      Il manque la variable d'environnement "GITHUB_OAUTH_APP_CLIENT_SECRET".
+    `)
+    process.exit(1);
   }
+}
 
-  const urlGithubOAuth =
-    `https://github.com/login/oauth/access_token?code=${code}&client_id=${client_id}&client_secret=${client_secret}`
-
-  got.post(urlGithubOAuth, { json: true }).then(githubResponse => {
-    const access_token = new URLSearchParams(githubResponse.body).get('access_token')
-
-    if(!access_token){
-      res.status(400)
-        .header('Content-Type', 'text/html')
-        .send(htmlTemplate`
-          <h1>Erreur</h1>
-          <p>le <code>access_token</code> attendu de la part de Github n'a pas été récupéré</p>
-          <p>Peut-être que le fonctionnement de l'API github a changé ou que le code ne le trouve pas au bon endroit</p>
-        `)
-      return;
-    }
-
-    redirectUrl.searchParams.set(`access_token`, access_token)
-    res.redirect(302, redirectUrl.toString())
-  })
-})
+const githubClientId = process.env.GITHUB_OAUTH_APP_CLIENT_ID || ""
+const githubClientSecret = process.env.GITHUB_OAUTH_APP_CLIENT_SECRET || ""
+const gitlabClientId = process.env.GITLAB_OAUTH_APP_CLIENT_ID || ""
+const gitlabClientSecret = process.env.GITLAB_OAUTH_APP_CLIENT_SECRET || ""
+const port = process.env.PORT || 4000
+const host = process.env.HOST || 'localhost'
+const serverBaseUrl = process.env.SERVER_BASE_URL || "http://localhost:4000"
```

Comment:
I think it's fine to not have a default value here (and an error message if it's not set). An env variable is easy to setup

Vocabulary nit: A protocol+hostname+port triplet is called an ["origin"](https://developer.mozilla.org/en-US/docs/Glossary/Origin) and i think the env var name should reflect that (even if it's not super-known)